### PR TITLE
feat: register url prefix sampler for flamingo, resolves #17

### LINKED ISSUE #17: Usage of samplers
Flamingo core has hardcoded opencensus samplers that are not supported by the [bridge](https://pkg.go.dev/go.opentelemetry.io/otel/bridge/opencensus#hdr-Limitations).

The Flamingo samplers have been re-implemented in this module but have not been used so far.

Adding this module to a Flamingo app should replace the samplers in Handler and prefixrouter with the samplers from this module.

### PR DESCRIPTION
**Pull request by tessig**
closes #17

### CODE REVIEW

**Comment by tessig** [comment]
waits for https://github.com/i-love-flamingo/flamingo/pull/442

File: Readme.md [hunk 1]
```diff
@@ -27,20 +27,28 @@ Please note the limitations of the bridge: https://pkg.go.dev/go.opentelemetry.i
...
+| `flamingo.opentelemetry.otlp.http.enable`          | `false`                              | enables the OTLP HTTP exporter                                               |
+| `flamingo.opentelemetry.otlp.http.endpoint`        | `http://localhost:4318/v1/traces`    | URL to the OTLP collector                                                    |
+| `flamingo.opentelemetry.otlp.grpc.enable`          | `false`                              | enables the OTLP gRPC exporter                                               |
+| `flamingo.opentelemetry.otlp.grpc.endpoint`        | `grpc://localhost:4317/v1/traces`    | URL to the OTLP collector                                                    |
+| `flamingo.opentelemetry.tracing.sampler.allowlist` | `[]`                                 | list of URL paths that are sampled; if empty, all paths are allowed          |
+| `flamingo.opentelemetry.tracing.sampler.blocklist` | `[]`                                 | list of URL paths that are never sampled                                     |
```
Comment (carstendietrich) [inline Readme.md:51]: publicEndpoint config not documented..

File: sampler.go [hunk 2]
```diff
@@ -1,73 +1,140 @@
...
 
 type ConfiguredURLPrefixSampler struct {
-	Allowlist            config.Slice
-	Blocklist            config.Slice
-	IgnoreParentDecision bool
+	allowlist []string
+	blocklist []string
+}
+
+type SpanKindBasedSampler struct {
```
Comment (carstendietrich) [inline sampler.go:19]:
```suggestion
// AlwaysSampleSpanKindClient enforces sampling of outgoing http requests (client)
type AlwaysSampleSpanKindClient struct {
```

File: sampler.go [hunk 3]
```diff
@@ -1,73 +1,140 @@
...
 
-	"flamingo.me/flamingo/v3/framework/config"
-	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
+
+	"flamingo.me/flamingo/v3/framework/config"
 )
 
 type ConfiguredURLPrefixSampler struct {
```
Comment (carstendietrich) [inline sampler.go:14]: lets make this private as well

File: sampler.go [hunk 4]
```diff
@@ -1,73 +1,140 @@
...
+
+		return tracesdk.SamplingResult{
+			Decision:   decision,
+			Tracestate: psc.TraceState(),
+		}
+	}
+
+	// empty allowed means all
+	sample := len(c.allowlist) == 0
+	// check allowed if len is > 0, and decide if we should sample
```
Comment (carstendietrich) [inline sampler.go:80]:
```suggestion
	// decide if we should sample based on allow list
```

File: sampler.go [hunk 5]
```diff
@@ -1,73 +1,127 @@
...
-type ConfiguredURLPrefixSampler struct {
-	Allowlist            config.Slice
-	Blocklist            config.Slice
-	IgnoreParentDecision bool
+type configuredURLPrefixSampler struct {
+	allowlist []string
+	blocklist []string
+}
+
+type alwaysSampleSpanKindClient struct {
```
Comment (carstendietrich) [inline sampler.go:18]:
```suggestion
// alwaysSampleSpanKindClient enforces sampling of outgoing http requests (client)
type alwaysSampleSpanKindClient struct {
```

**Review by carstendietrich — APPROVED** [review]
What a day to be alive 🎉

**Pull request merged**